CMakeLists.txt

```diff
@@ -96,6 +96,20 @@ if (CMAKE_SYSTEM_NAME MATCHES "CYGWIN")
    endif()
 endif()
 
+if (CMAKE_SYSTEM_NAME MATCHES "Windows")
+   if ((CMAKE_GENERATOR MATCHES "MinGW Makefiles") OR
+       (CMAKE_GENERATOR MATCHES "MSYS Makefiles") OR
+       (CMAKE_GENERATOR MATCHES "Unix Makefiles"))
+
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wa,-mbig-obj")
```

Comment:
Not sure if needed. Makes no difference in my case.

Reply:
It might be that the `set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wa,-mbig-obj")` is not needed anymore in combination with the `-O1` or `-Og`, but in the past you needed it and for Cygwin we also needed it in the past so I left it for compatibility.
I don't think it hurts, but I will have to do some tests ...

Reply:
I just finished a number of tests and it looks like the `set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wa,-mbig-obj")` is not necessary anymore.
The description in the assembler man page says:
```
       -mbig-obj
           On PE/COFF target this option forces the use of big object file format, which allows more than 32768 sections.
```
I don't think it hurts and might prevent some problems in the future, however an extra flag set is always a bit tricky.

@doxygen Should we leave the flag or should we remove it?

Reply:
@albert-github I think we should remove it if not needed.

Reply:
OK, I;ve removed it (but we should keep it in mind for the future).